### PR TITLE
make fill take function as value for matrix

### PR DESCRIPTION
**Pull request by Xuefeng-Zhu**

### CODE REVIEW

File: matrix.js [hunk 1]
```diff
@@ -234,14 +235,9 @@
...
     mean = mean || 0;
-    type = type || Float64Array;
-    var data = new type(i * j),
-        k;
-
-    for (k = 0; k < i * j; k++)
-      data[k] = deviation * Math.random() + mean;
-
-    return Matrix.fromTypedArray(data, [i, j]);
+    return Matrix.fill(count, function() {
```
Comment (bartvanandel) [inline matrix.js]: Undefined variable `count`

File: matrix.js [hunk 2]
```diff
@@ -189,9 +189,10 @@
 
     var size = i * j,
         data = new type(size),
+        isValueFn = typeof value === 'function',
         k;
     for (k = 0; k < size; k++)
-      data[k] = value;
+      data[k] = (isValueFn && value()) || value;
```
Comment (bartvanandel) [inline matrix.js]: Two issues with this approach.
1) When value() returns false, code will try to set data[k] the function instead.
2) Value function should probably take (row, col) as input to improve its usability

Summarized, better use something like this:
`      data[k] = isValueFn ? value(k / j | 0, k % j) : value;`

Note: adapted from `Matrix.prototype.map` which uses same approach for row/col computation.

Side note: for consistency of value names `fill` should probably use `r, c` as arguments: `i` and `j` are more often used as element indices, not size parameters.

Comment (Xuefeng-Zhu) [inline matrix.js]: updated

**Comment by mateogianolio** [comment]
See changes in #101.

**Pull request closed without merging**